src/lib.rs

```diff
@@ -46,16 +78,11 @@ impl Default for Decoder {
 }
 
 impl Decoder {
-    pub fn new() -> Self {
+    pub fn new_with_settings(settings: &Settings) -> Self {
```

Comment:
```suggestion
    pub fn with_settings(settings: &Settings) -> Self {
```

Seems to follow the Rust code style a bit better